crypto/stark/src/solidity_seralizer.rs

```diff
@@ -0,0 +1,356 @@
+use crate::{channel::*, constraints::Constraints, proof_of_work, Proof};
+use hex::encode;
+use std::{collections::BTreeMap, prelude::v1::*};
+use zkp_hash::Hash;
+use zkp_merkle_tree::{Commitment, Error as MerkleError};
+use zkp_primefield::FieldElement;
+use zkp_u256::U256;
+
+// This trait is a simple json encoding trait which exports the values
+// in the format expected by the ethereum call, the hand made version
+// is used instead of serde because it differ in important ways
+trait SoliditySeralize {
+    fn sol_encode(&self) -> String;
+}
+
+impl SoliditySeralize for FieldElement {
+    fn sol_encode(&self) -> String {
+        format!("\"0x{}\"", encode(self.as_montgomery().to_bytes_be()))
+    }
+}
+
+impl SoliditySeralize for Hash {
+    fn sol_encode(&self) -> String {
+        format!("\"0x{}\"", encode(self.as_bytes()))
+    }
+}
+
+impl SoliditySeralize for Commitment {
+    fn sol_encode(&self) -> String {
+        format!("\"0x{}\"", encode(self.hash().as_bytes()))
+    }
+}
+
+// This recursive call makes a javascript array, for any type which can be
+// serialized
+impl<T: SoliditySeralize> SoliditySeralize for Vec<T> {
+    fn sol_encode(&self) -> String {
+        let mut res: String = "".to_string();
+        if self.is_empty() {
+            return res;
+        }
+
+        let (last, rest) = self.split_last().unwrap();
+        res.push_str(&"[\n");
+        for data in rest.iter() {
+            res.push_str(&format!("{},\n", data.sol_encode()));
+        }
+        res.push_str(&format!("{}\n", last.sol_encode()));
+        res.push_str("]");
+        res
+    }
+}
+
+// TODO - Make this function smaller
```

Comment:
Yeah, the problem is that `Proof` has no structure. This needs to change firsts. Out of scope for this PR though.